packages/invoice-template/src/common/LabeledField/index.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { ReactNode, FunctionComponent } from "react";
+import styled from "styled-components";
+
+type Orientation = "horizontal" | "vertical";
+
+type LabeledFieldProps = {
+    orientation: Orientation;
+    label: ReactNode;
+    value: ReactNode;
+    margin?: string;
+};
+
+const LabeledField: FunctionComponent<LabeledFieldProps> = ({ orientation, label, value, margin }) => (
+    <Root margin={margin} orientation={orientation}>
+        <Label>{label}</Label>
+        <Value orientation={orientation}>{value}</Value>
+    </Root>
+);
+
+type OrientationProps = {
+    orientation: Orientation;
+};
+
+type RootProps = {
+    margin?: string;
+} & OrientationProps;
+
+const Root = styled.div<RootProps>`
+    display: flex;
+    flex-direction: ${props => (props.orientation === "horizontal" ? "row" : "column")};
```

Comment:
TBH I still don't like this pattern I feel that destructuring is far more concise and easier to read. Do you have any opinion on that?